external/loaders/loaders/mappers/_hybrid.py

```diff
@@ -1,88 +1,179 @@
+def apparent_moistening(data: FineResBudget):
+    eddy_flux = eddy_flux_coarse(
+        data.eddy_flux_vulcan_omega_sphum,
+        data.sphum_vulcan_omega_coarse,
+        data.vulcan_omega_coarse,
+        data.sphum,
+    )
+    eddy_flux_convergence = convergence(eddy_flux, data.delp, dim="pfull")
+    return (
+        (data.qv_dt_fv_sat_adj_coarse + data.qv_dt_phys_coarse + eddy_flux_convergence)
+        .assign_attrs(
+            units="kg/kg/s",
+            long_name="apparent moistening from high resolution data",
+            description=(
+                "Apparent moistening due to physics and sub-grid-scale advection. "
+                "Given by "
+                "sat adjustment (dycore) + physics tendency + eddy-flux-convergence"
+            ),
         )
+        .rename("Q2")
+    )
+
 
-    def __len__(self):
-        return len(self.keys())
+def _open_fine_res_dataset(
+    # created by this commit:
+    # https://github.com/VulcanClimateModeling/vcm-workflow-control/commit/3c852d0e4f8b86c4e88db9f29f0b8e484aeb77a1
+    # I manually consolidated the metadata with zarr.consolidate_metadata
+    fine_url: str = "gs://vcm-ml-experiments/default/2021-04-27/2020-05-27-40-day-X-SHiELD-simulation/fine-res-budget.zarr",  # noqa: E501
+    # created by this commit
+    # https://github.com/VulcanClimateModeling/vcm-workflow-control/commit/dd4498bcf3143d05095bf9ff4ca3f1341ba25330
+    nudge_url="gs://vcm-ml-experiments/2021-04-13-n2f-c3072/3-hrly-ave-rad-precip-setting-30-min-rad-timestep-shifted-start-tke-edmf",  # noqa: E501
+) -> xarray.Dataset:
 
-    def __iter__(self):
-        return iter(self.keys())
+    fine = open_zarr(fine_url, consolidated=True)
+    # compute apparent sources
+    fine["Q1"] = apparent_heating(fine)
+    fine["Q2"] = apparent_moistening(fine)
+    # shift the data to match the other time series
+    fine_shifted = fine.assign(time=fine.time - timedelta(minutes=7, seconds=30))
+
+    nudge_physics_tendencies = open_zarr(
+        nudge_url + "/physics_tendencies.zarr", consolidated=True
+    )
+    nudge_state = open_zarr(nudge_url + "/state_after_timestep.zarr", consolidated=True)
+    nudge_tends = open_zarr(nudge_url + "/nudging_tendencies.zarr", consolidated=True)
+
+    merged = xarray.merge(
+        [gfdl_to_standard(fine_shifted), nudge_state, nudge_physics_tendencies],
+        join="inner",
+    )
+
+    # dQ1,2,u,v
+    # "hybrid" definitions for humidity and moisture
+    merged["dQ1"] = (
+        merged["Q1"] - merged["tendency_of_air_temperature_due_to_fv3_physics"]
+    )
+    merged["dQ2"] = (
+        merged["Q2"] - merged["tendency_of_specific_humidity_due_to_fv3_physics"]
+    )
+    merged["dQxwind"] = nudge_tends.x_wind_tendency_due_to_nudging
+    merged["dQywind"] = nudge_tends.y_wind_tendency_due_to_nudging
+
+    # Select the data we want to return
+    return (
+        xarray.merge(
+            [
```

Comment:
Not all input variables are in this list, but there should be enough to compute cos zenith angle and the surface geo-potential.

Reply:
why not just return `merged`?